### PR TITLE
Meistejw/feature/013/add dialogue to add case

### LINKED ISSUE #13: Add button to enable adding a case
Health services needs to be able to add a positive case through the frontend.

Acceptance Criteria:

- [x] Button for adding a case on the person profile
- [x] Opens a dialog asking for the test date
- [x] New case appears in the frontend as well as the db

### PR DESCRIPTION
**Pull request by meistejw**
Adds a dialog box that opens up and allows you to select the test date of a user when adding them as a positive case.

closes #13

### CODE REVIEW

File: src/app/add-case-dialogue/add-case-dialogue.component.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<h1 mat-dialog-title>Add Positive Case</h1>
+<div mat-dialog-content>
+  <p>What is the date of the person's test?</p>
+  <mat-form-field appearance="fill">
+    <mat-label>Choose a date</mat-label>
+    <input matInput [matDatepicker]="picker" [(ngModel)]="data.testDate">
```
Comment (noahthomas88) [inline src/app/add-case-dialogue/add-case-dialogue.component.html:6]: Interesting, so this matDatepicker is internal to angular?

Comment (meistejw) [reply src/app/add-case-dialogue/add-case-dialogue.component.html:6]: It comes from Material, so I don't really know if you would consider that internal to angular or not.

File: src/app/app.module.ts [hunk 2]
```diff
@@ -33,9 +37,15 @@ import { PersonProfileComponent } from './person-profile/person-profile.componen
...
+    HttpClientModule,
+    FormsModule,
+    MatDatepickerModule,
+    MatNativeDateModule
   ],
   providers: [PositiveCaseService, UserService],
-  bootstrap: [AppComponent]
+  bootstrap: [AppComponent],
+  entryComponents: [AddCaseDialogueComponent],
+  schemas: [CUSTOM_ELEMENTS_SCHEMA]
```
Comment (noahthomas88) [inline src/app/app.module.ts:49]: I don't understand where this CUSTOM_ELEMENTS_SCHEMA is used. Did I miss it somewhere?

**Review by noahthomas88 — APPROVED** [review]
Great work Jake!

File: src/app/app.module.ts (hunk 2 above)
Comment (meistejw) [reply src/app/app.module.ts:49]: It is declared in the schema section of app.module.ts. It is necessary for some Material html tags in the add-positive-case component.

File: src/app/add-case-dialogue/add-case-dialogue.component.html (hunk 1 above)
Comment (DRSisco) [reply src/app/add-case-dialogue/add-case-dialogue.component.html:6]: Nope. It's a wonderful add on!

However most people seem to prefer bootstrap, so you'll see that too.

Though for stringing up a quick app, especially when you don't know angular... angular materials is really good at teaching and being easy to built cool things.

**Pull request merged**